### PR TITLE
Conservatively handle `foreigncall` exprs

### PR DESCRIPTION
**Pull request by TH3CHARLie**

### CODE REVIEW

**Comment by TH3CHARLie** [comment]
~~test is broken at dispatch check on `length`. I saw we have a few other `length` usages across `find_escapes`, not sure why this breaks test.~~

I think it's because `args[3]` is `Any` type, I tried to convert it into `SimpleVector` type but encountered import error.

**Comment by TH3CHARLie** [comment]
With this, we can eliminate the error in #25 
```
julia> @analyze_escapes plot(rand(10, 3))
typeof(plot)(_2::Tuple{Matrix{Float64}} →)
1 → 1 ─ %1 = Core.getfield(_2, 1)::Matrix{Float64}
2 ↑ │   %2 = invoke Plots.:(var"#plot#159")($(QuoteNode(Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}()))::Base.Pairs{Symbol, V, Tuple{Vararg{Symbol, N}}, NamedTuple{names, T}} where {V, N, names, T<:Tuple{Vararg{Any, N}}}, RecipesBase.plot::typeof(plot), %1::Any)::Plots.Plot
3 ◌ └──      return %2
```

File: src/EscapeAnalysis.jl [hunk 1]
```diff
@@ -14,6 +14,7 @@ import Core:
     ReturnNode,
     GotoNode,
     GotoIfNot
+    SimpleVector
```
Comment (aviatesk) [inline src/EscapeAnalysis.jl]:
```suggestion
    GotoIfNot,
    SimpleVector
```

syntax error

File: src/EscapeAnalysis.jl [hunk 2]
```diff
@@ -251,6 +252,12 @@ function find_escapes(ir::IRCode, nargs::Int)
                     if isa(lhs, GlobalRef)
                         add_change!(rhs, ir, Escape(), changes)
                     end
+                elseif head === :foreigncall
+                    # for foreigncall we simply escape every argument and gc-root
+                    # TODO: we can apply similar strategy like builtin calls
+                    #       to specialize some foreigncalls
+                    foreigncall_nargs = length(SimpleVector{Type}(stmt.args[3]))
```
Comment (aviatesk) [inline src/EscapeAnalysis.jl]:
```suggestion
```

**Pull request merged**